lcls_tools/common/devices/yaml/generate.py

```diff
@@ -134,14 +134,17 @@ def _construct_pv_list_from_control_system_name(
         # Use the control system name to get all PVs associated with device
         pv_dict = {}
         for search_term, handle in search_with_handles.items():
+            field = str()
+            if "." in search_term:
+                search_term, field = search_term.split(".")
             # End of the PV name is implied in search_term
             try:
                 pv_list = meme.names.list_pvs(name + ":" + search_term, sort_by="z")
                 # We expect to have ZERO or ONE result returned from meme
                 if pv_list != list():
                     if len(pv_list) == 1:
                         # get the pv out of the results
-                        pv = pv_list[0]
+                        pv = f"{pv_list[0]}.{field}" if field else pv_list[0]
```

Comment:
Is field strictly evaluated as `False` in this case (`str()`)? Maybe we can set field to `False` initially?

Reply:
do we need the string initialization here?